admin/app/views/layouts/solidus_admin/application.html.erb

```diff
@@ -5,6 +5,7 @@
   </head>
   <body>
     <h1 class="text-4xl">Layout is rendered</h1>
+    <%= render component("main_nav").new %>
```

Comment:
Hey @elia, I pointed out some of the befits and features that dry-system provides in [another answer](https://github.com/solidusio/solidus/pull/5099#discussion_r1217434038).

> <%= render component("main_nav").new(items: SolidusAdmin::NavItem.all) %>

I think it's good having the default value for `items` already defined on the initializer of `MainNavComponent`. In this case, it doesn't make such a difference, but maybe other components will be reused, which would avoid boilerplate code. In that situation, dry-system's [auto_inject](https://dry-rb.org/gems/dry-system/1.0/dependency-auto-injection/) would also help.

> class SolidusAdmin::NavItem < ActiveModel::Model
>  def self.all
>    @all ||= SolidusAdmin::Config.nav_items.map { new _1 }
>  end
> end

I know that's not a formal proposal, but regardless of using dry-system or not, I'd go with a PORO for nav item entities. In the end, we don't need to deal with validations, which would be a case for `ActiveModel::Model`.

> SolidusAdmin::Config.nav_items += [
>  { name: "Extra item", icon: :gift, … }
> ]

I'd also like to avoid having `SolidusAdmin::Config` as a hodgepodge. For anything else that configuring simple primitives, I think it'd be good having a richer interface, whether that one is provided by dry-system or something we own.

Reply:
@elia, please, take a look at the last commit I pushed (__Auto-inject dependencies into components__). I also included [`dry-sytem`'s auto_inject](https://dry-rb.org/gems/dry-system/1.0/dependency-auto-injection/) within this spike, to better showcase both how having defaults defined in the `#initialize` method, and having nav items registered in the container help with avoiding boilerplate. Also pinging @kennyadsl.